src/nanoarrow/array.c

```diff
@@ -910,10 +910,6 @@ ArrowErrorCode ArrowArrayViewValidateFull(struct ArrowArrayView* array_view,
                                           struct ArrowError* error) {
   for (int i = 0; i < 3; i++) {
     switch (array_view->layout.buffer_type[i]) {
-      case NANOARROW_BUFFER_TYPE_UNION_OFFSET:
```

Comment:
Hmm, dense union offsets don't have to be increasing?

Reply:
Ah, "The respective offsets for each child value array must be in order / increasing." So if we want to validate this it'll be trickier.